### DISCUSSION TITLE
插件弹出窗口中增加其它右键菜单功能

### DISCUSSION THREAD

**Discussion by oott123** [chunibyocola/sc-translator-crx#48] [Ideas]
![image](https://user-images.githubusercontent.com/905663/172969160-7a607411-f2ba-4a8f-9ab8-8f8c551f90b2.png)

这里能不能把这些功能加进去呢，比如打开独立窗口或者翻译整页：

![image](https://user-images.githubusercontent.com/905663/172969188-cbeb3969-36d7-4751-b505-bfb828da6d57.png)

因为右键菜单如果太多会被折叠，一些低频功能为了日常使用方便，还是在插件弹出窗口中比较好用

**Comment by chunibyocola** [comment]
右键菜单这几个功能怎么说呢，只有“翻译当前页面”和“用 PDF 打开”能找到适合图标。再就是空间不是那么充足，我觉得现在都有点多了，不那么美观😂。

再就是“翻译当前页面”、“打开独立翻译窗口”、“翻译你选择的文本”和“朗读你选择的文本”都是可以用快捷键来激活的，通过快捷键的方式能更方便，我是这么想的。

**Reply by oott123** [reply]
快捷键对于鼠标党来讲不够直观而且根本记不住……不如考虑折叠？或者做到扩展图标右键上（这是可以自定义的吗我不太清楚）？

**Reply by chunibyocola** [reply]
> 快捷键对于鼠标党来讲不够直观而且根本记不住……不如考虑折叠？

折叠是指哪方面的折叠呢，折叠图标的话就失去用两个步骤就能做到的便利性了。

> 或者做到扩展图标右键上

这个是可以的，只是很少有用户会右键点击扩展图标我没太关注这里，下个版本我给“翻译当前页面”加上吧。

**Reply by oott123** [reply]
> 折叠是指哪方面的折叠呢

做成汉堡菜单那种折叠，就像 github 这样
![image](https://user-images.githubusercontent.com/905663/172975639-8067f3c5-3fe6-445b-bee7-973a8aba8d32.png)


> 下个版本我给“翻译当前页面”加上吧。

其实我是想要 “打开独立翻译窗口” ……

**Reply by chunibyocola** [reply]
> 做成汉堡菜单那种折叠，就像 github 这样

这样同样是失去一定的便利性了。

> 其实我是想要 “打开独立翻译窗口” ……

~~你这样说我才发现 Firefox 的扩展右键菜单不支持这个功能，因为“打开独立翻译窗口”是本来就支持右键扩展图标的。~~

我理解了，你是想要右键菜单不折叠，能直接点击网页翻译，而有时又需要独立翻译窗口。还是推荐使用快捷键，单独记住这一个就可以了，我自己的话是设置成`ALT + B`，同时设置为全局快捷键（可惜 Firefox 也不支持）。

**Reply by oott123** [reply]
![image](https://user-images.githubusercontent.com/905663/172988959-ee3d3edf-77d0-4def-9002-851d0a2abac1.png)

但是有的扩展(ublock origin)确实能显示

![image](https://user-images.githubusercontent.com/905663/172988992-124cd8cf-c14b-43a8-a924-5f05e55dcd7c.png)

Sc 就啥也没有……



我看 [文档](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/menus/ContextType)，按说 browser_action 里面是有的啊，是不是和 #47 一个问题呢？

**Reply by chunibyocola** [reply]
我想起来了，以前的版本在 Firefox 也是有的，时间太长我给忘了...已确认是和 #47 是一个 bug 😂。

**Reply by chunibyocola** [reply]
已更新到 v4.1.2，更新一下就能用扩展图标右键菜单打开独立翻译窗口了。因为我日常是不用 Firefox 的，又由于 Firefox 暂时还不支持 MV3 的标准，有一些东西根本注意不到，非常感谢！🙇‍

**Comment by oott123** [comment]
![image](https://user-images.githubusercontent.com/905663/173030912-37f973b3-ef3f-46ee-b969-d6e80156f529.png)
酷
